translator/writer/prim_writer.cpp

```diff
@@ -359,12 +361,16 @@ const UsdArnoldPrimWriter::ParamConversion* UsdArnoldPrimWriter::getParamConvers
  **/ 
 void UsdArnoldPrimWriter::writeNode(const AtNode *node, UsdArnoldWriter &writer)
 {
-    // we're exporting a new node, 
-    // se we can clear the list of already exported attributes
+    // we're exporting a new node, so we store the previous list of exported 
+    // attributes and we clear it for the new node being written
+    std::unordered_set<std::string> prevExportedAttrs = _exportedAttrs;
     _exportedAttrs.clear();
 
     // Now call the virtual function write() defined for each primitive type
     write(node, writer); 
+
+    // restore the previous list (likely empty, unless there have been recursive creation of nodes)
+    _exportedAttrs = prevExportedAttrs;
```

Comment:
The assignment operator and the constructor of the `std::unordered_map` copies elements, so we would end up duplicating elements without actually needing them. A more optimal solution is to use swap, which only swaps internal pointers between two `std::unordered_map`, instead of moving elements.

[Reference](https://en.cppreference.com/w/cpp/container/unordered_map/swap)

> Exchanges the contents of the container with those of other. Does not invoke any move, copy, or swap operations on individual elements.

So something like this:

```cpp
decltype(_exportedAttrs) prevExportedAttrs;
prevExportedAttrs.swap(_exportedAttrs);
write(node, writer);

_exportedAttrs.swap(prevExportedAttrs);
```

Bonus, using `decltype` lets you to change the stored type in the `std::unordered_set`, without breaking the code. :)